src/cdk/datetime/datepicker-input.ts

```diff
@@ -0,0 +1,345 @@
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  OnDestroy,
+  Optional,
+  Output,
+} from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import {DateAdapter} from './date-adapter';
+import {Subscription} from 'rxjs';
+import {CdkDatepicker} from './datepicker';
+import {CDK_DATE_FORMATS, CdkDateFormats} from './date-formats';
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * Provider that allows the datepicker to register as a ControlValueAccessor.
+ */
+export const DATEPICKER_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * An event used for datepicker input and change events. For consistency, we always use
+ * DatepickerInputEvent instead.
+ */
+export class DatepickerInputEvent<D> {
+  /** The new value for the target datepicker input. */
+  value: D | null;
+
+  constructor(
+      /** Reference to the datepicker input component that emitted the event. */
+      public target: CdkDatepickerInput<D>,
+      /** Reference to the native input element associated with the datepicker input. */
+      public targetElement: HTMLElement) {
+    this.value = this.target.value;
+  }
+}
+
+
+/** Directive used to connect an input to a CdkDatepicker. */
+@Directive({
+  selector: 'input[cdkDatepicker]',
+  providers: [
+    DATEPICKER_VALUE_ACCESSOR,
+    DATEPICKER_VALIDATORS,
+  ],
+  host: {
+    '[attr.aria-owns]': '(_datepicker.id) || null',
+    '[attr.min]': 'min ? _dateAdapter.toIso8601(min) : null',
+    '[attr.max]': 'max ? _dateAdapter.toIso8601(max) : null',
+    '[disabled]': 'disabled',
+    '(input)': '_onInput($event.target.value)',
+    '(change)': '_onChange()',
+    '(blur)': '_onBlur()',
+  },
+  exportAs: 'cdkDatepickerInput',
+})
+export class CdkDatepickerInput<D> implements AfterContentInit, ControlValueAccessor, OnDestroy,
+  Validator {
+  /** The datepicker that this input is associated with. */
+  @Input('cdkDatepicker')
+  set datepicker(value: CdkDatepicker<D>) {
+    this._registerDatepicker(value);
+  }
+  _datepicker: CdkDatepicker<D>;
+
+  /** Registering datepicker with this input. */
+  private _registerDatepicker(value: CdkDatepicker<D>) {
+    if (value) {
+      this._datepicker = value;
+      this._datepicker._registerInput(this);
+    }
+  }
+
+  /** Function that can be used to filter out dates within the datepicker. */
+  @Input()
+  set cdkDatepickerFilter(value: (date: D | null) => boolean) {
```

Comment:
The property here should just be `filter` with an alias for the input: `@Input('cdkDatepickerFilter')`